packages/ra-core/src/form/useUnique.spec.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import {
+    Basic,
+    DataProviderErrorOnValidation,
+    WithAdditionalFilters,
+    WithCustomMessage,
+} from './useUnique.stories';
+
+describe('useUnique', () => {
+    it('should show the default error when a field is not unique', async () => {
+        render(<Basic />);
+
+        await screen.findByDisplayValue('John Doe');
+
+        fireEvent.click(screen.getByText('Submit'));
+
+        await screen.findByText('Must be unique');
+    });
+
+    it('should not show the default error when a field is unique', async () => {
+        window.alert = jest.fn();
+        render(<Basic />);
+
+        await screen.findByDisplayValue('John Doe');
+        fireEvent.change(screen.getByDisplayValue('John Doe'), {
+            target: { value: 'Jordan Doe' },
+        });
+
+        fireEvent.click(screen.getByText('Submit'));
+
+        expect(screen.queryByText('Must be unique')).toBeNull();
```

Comment:
Not sure this test fails for the right reasons, as the notifications are async

Reply:
Done